ports/atmel-samd/common-hal/busio/UART.c

```diff
@@ -150,7 +150,7 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
         self->buffer = (uint8_t *) gc_alloc(self->buffer_length * sizeof(uint8_t), false, true);
```

Comment:
Why not use m_malloc here?  You'd get the error message for free.

Reply:
So that we can deinit the UART before throwing the error.